File: Losser/FrobNormLosser.jl
module FrobNormLosserModule
include("./Losser.jl")

import .LosserModule: computeLoss, Homography
import LinearAlgebra

#Extends the computeLoss function from LosserModule
function computeLoss(H_true::Homography, H_computed::Homography)::Float64
    return LinearAlgebra.norm(H_true.H - H_computed.H)
end
end

File: src/HomoTopiContinuation/Losser/FrobNormLosser.py
from HomoTopiContinuation.Losser.Losser import Losser
from HomoTopiContinuation.DataStructures.datastructures import Homography
import numpy as np


class FrobNormLosser(Losser):
    """
    Class to compute the loss between two homographies using the Frobenius norm of the difference.
    """

    def computeLoss(h_true: Homography, h_computed: Homography) -> float:
        """
        Compute the loss between two homographies using the Frobenius norm of the difference.

        Args:
            h_true (Homography): The true homography
            h_computed (Homography): The computed homography

        Returns:
            float: The loss between the two homographies
        """
        return np.linalg.norm(h_true() - h_computed(), 'fro')
